### PR TITLE
global: integration of invenio-files-rest

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Refactorices the code in order to make it work with
  invenio-files-rest and invenio-records.

* Adds a macro to generate the iframe which shows the content of the
  file.

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

**Comment by JavierDelgadoFernandez** [comment]
Addressed in inveniosoftware/invenio-files-rest#8.

**Comment by JavierDelgadoFernandez** [comment]
### Working on:
- [ ] Test coverage issue (Test still running over more than 90% of the code but the detector is not working properly).
- [ ] Example app.

**Comment by JavierDelgadoFernandez** [comment]
I have remove one test module (``test_invenio_previewer_ext``) since it was checking the same behaviour of ``test_invenio_previwer_views`` and due to the integration of ``invenio_files_rest``, the test cases were becoming exactly the same.

File: tests/test_invenio_previewer_views.py [hunk 1]
```diff
@@ -28,147 +28,182 @@
...
+    thread = Process(target=run_server)
+    thread.start()
+    while True:
+        try:
+            response = requests.get(url, stream=True, timeout=0.5)
+            thread.terminate()
+            thread.join()
+            return response
+        except requests.exceptions.ConnectionError:
+            pass
```
Comment (JavierDelgadoFernandez) [inline tests/test_invenio_previewer_views.py]: Although I am planing to use a counter in the ``while True`` loop, the method will still looking a bit ugly. I haven't found a better solution to solve the issue of nested HTTP requests. Is this solution ok?

**Comment by lnielsen** [comment]
What is the status of this PR.

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen The issue with the code detection stills there so the code can be reviewed but the tests are going to be changed (Not the tests per se but the way to run the server and check the response)

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen The test are finally passing and the module is ready from my side. Could you take a look, please?

File: invenio_previewer/extensions/csv_dthreejs.py [hunk 2]
```diff
@@ -24,42 +24,45 @@
...
-                universal_detector.feed(
-                    dialect.delimiter.join(row).encode('utf-8'))
-            is_valid = True
-        except csv.Error as e:
-            current_app.logger.debug(
-                'File %s is not valid CSV: %s' % (document.get_filename(), e))
-        finally:
-            universal_detector.close()
+    fp = file.open()
+    content = fp.read().decode('utf-8')
```
Comment (lnielsen) [inline invenio_previewer/extensions/csv_dthreejs.py]: Only read X bytes

File: invenio_previewer/extensions/csv_dthreejs.py [hunk 3]
```diff
@@ -67,17 +70,17 @@ def validate_csv(document):
     }
 
 
-def can_preview(document):
+def can_preview(file):
     """Determine if the given file can be previewed."""
-    if document.extension == 'csv':
-        return validate_csv(document)['is_valid']
+    if file.file['key'].endswith('.csv'):
```
Comment (lnielsen) [inline invenio_previewer/extensions/csv_dthreejs.py]: Probably check if ``local`` is True and use ``file['uri']``

**Comment by JavierDelgadoFernandez** [comment]
@lnielsen Updated.

WRT zenodo/zenodo#465, now it is showing an alert:
lars

**Comment by lnielsen** [comment]
Great :-)

**Pull request merged**